pkg/json/parse.go

```diff
@@ -145,38 +148,54 @@ func (ing *Parser) parseObject(ctx parserContext, input *jsonom.Object) (*Parsed
 			if !f(schNode, keyValue.Value()) {
 				continue
 			}
+			if discrims, cached := ing.discriminator[ctx.schemaNode]; cached {
+				for _, snode := range discrims {
+					kv := input.Get(ls.AsPropertyValue(snode.GetProperty(ls.AttributeNameTerm)).AsString())
+					newCtx := ctx
+					newCtx.schemaNode = snode
+					newCtx.path = newCtx.path.AppendString(kv.Key())
+					_, err := ing.parseDoc(newCtx, kv.Value())
+					if err != nil {
+						return err
+					}
+				}
+			}
 
-			newCtx := ctx
-			newCtx.path = newCtx.path.AppendString(keyValue.Key())
-			newCtx.schemaNode = schNode
+			// check if discrimator term exists, terminate early if none found
+			if schNode.HasLabel(ls.TypeDiscriminatorTerm) || schNode.HasLabel("typeDiscriminator") {
```

Comment:
This is not right. The ing.discriminator should keep all schema nodes under ctx.schemaNode that has discriminators. So the cache should be initialized to do that. If ing.schemaNode[ctx.schemaNode] is nll. iterate all nextNodes, and put the ones that have typeDiscriminator term into the map.